### PR TITLE
Skip the line in Dockerfile if it is a comment

### LINKED ISSUE #2: Support for modifying labels
With the `structure` property it is quite easy to modify most values, by first finding the value to modify, then replacing the lines (known from `startline` and `endline`) in the `lines` property.

For example, for CMD:
```
for insn in parser.structure:
  if insn['instruction'] == 'CMD':
    content = 'CMD my-new-cmd-value\n'
    lines = parser.lines
    del lines[insn['startline']:insn['endline'] + 1]
    lines.insert(startline, content)
    parser.lines = lines
    break
```

However, for the LABEL instruction it is much more complicated as firstly there are two forms of LABEL, secondly there is quoting to deal with, and thirdly a single LABEL instruction can contain many labels.

It would be nice to provide support for modifying a single label, handling all these cases.

### PR DESCRIPTION
**Pull request by vrutkovs**
Fixes #2

### CODE REVIEW

**Comment by twaugh** [comment]
I would like more tests, so that we know how these sorts of things behave:

```
# ignore this \
RUN but run this
```

```
RUN run some stuff && \
    # ignore this, but \
    run some other stuff
```

For the record, I think the docker parser's behaviour here is terrible, but we have to do what it does.

File: tests/test_parser.py [hunk 1]
```diff
@@ -52,7 +52,12 @@ def test_dockerfile_structure(self, dfparser):
                           " label  foo  \\\n",  # extra ws
                           "    # comment\n",    # should be ignored
                           "    bar  \n",        # extra ws, continuation line
-                          "USER  {0}".format(NON_ASCII)]   # extra ws, no newline
+                          "USER  {0}\n".format(NON_ASCII),
+                          "# comment \\\n",    # extra ws
```
Comment (twaugh) [inline tests/test_parser.py]: How about adding a test with extra whitespace *after* the backslash as well? (You'll be surprised what docker does with that...)

Comment (twaugh) [inline tests/test_parser.py]: Also how about two backslashes at the end of the line?

**Comment by vrutkovs** [comment]
Implemented those, I really hope Docker's parser won't crash on those cases

**Comment by twaugh** [comment]
Is this still current? I see #2 was marked as being closed by #4.

**Comment by vrutkovs** [comment]
It is, it fixes several other cases (e.g. comment in multi-line instruction)
